Mathlib/Init/Data/Nat/Lemmas.lean

```diff
@@ -209,20 +213,24 @@ protected theorem one_le_bit0 : ∀ n : ℕ, n ≠ 0 → 1 ≤ bit0 n
 end bit
 
 /- successor and predecessor -/
-
+/--`discriminate` is basically the principle of mathematical induction-/
 def discriminate (H1 : n = 0 → α) (H2 : ∀m, n = succ m → α) : α :=
   match n with
   | 0 => H1 rfl
   | succ m => H2 m rfl
 
 lemma one_eq_succ_zero : 1 = succ 0 := rfl
 
+/--`two_step_induction`: If `P` is a property of natural numbers
+that is true for `0` and `1` and it it is true that `P m` and `P (m+1)`
+implies `P m+2`, then `P a` holds for all `a : N`-/
```

Comment:
Please remove this comment, it doesn't belong in this PR

Reply:
Done